src/sagemaker/workflow/function_step.py

```diff
@@ -225,6 +228,10 @@ def properties(self):
     def to_request(self) -> RequestType:
         """Gets the request structure for workflow service calls."""
         request_dict = super().to_request()
+
+        if self.is_final_step:
```

Comment:
I dont see any unit tests using the @step wrapper that hits this section of code. FunctionSteps are created when users use this annotation. Did we test this / plan to test it? My intuition is that this won't work as they have a different critical path.

Reply:
The unit test using the @step decorator is in tests/unit/sagemaker/workflow/test_function_step.py, where I used a current case and defined the value of is_final_step as true and test its result after the step is created.
<img width="1077" alt="image" src="https://github.com/user-attachments/assets/35993782-e595-4e97-acad-08b1920ad23f">